libs/designer/src/lib/core/utils/parameters/dynamicdata.ts

```diff
@@ -320,6 +323,20 @@ export async function getFolderItems(
     );
 
     return getLegacyDynamicTreeItems(connectionId, connectorId, operationId, inputs, filePickerInfo, managedIdentityRequestProperties);
+  } else if (isDynamicTreeExtension(definition) && filePickerInfo) {
+    const { open, browse } = filePickerInfo;
+    const { connectorId } = operationInfo;
+    const connectionId = connectionReference?.connection.id as string;
+    const { operationId, parameters } = selectedValue ? browse : open;
+
+    const operationParameters = getParameterValuesForDynamicInvoke(parameters as DynamicParameters, nodeInputs, idReplacements);
+
+    let dynamicState = definition.extension.dynamicState;
+    if (selectedValue) {
+      dynamicState = { ...dynamicState, selectionState: selectedValue.selectionState };
```

Comment:
When we implement this extension for LA connectors, this code will be removed because you are not relying on the selectedItemValuePath property of the reference parameters which tells which property to extract from the selectionState and that should be sent in parameters today. Your service should be reading that parameter and modify your  api call if these need to be sent in dynamicState property.

If you do not want to change it now, it will be breaking change for you then.

Reply:
Thanks Priti! We are okay with taking the breaking change in if it's okay to leave this like this for now. We'll fast-follow next week to use/pass selectedItemValuePath correctly instead.

Reply:
Created this PBI for fast-follow next week:
[Product Backlog Item 21887309](https://msazure.visualstudio.com/One/_workitems/edit/21887309): [Fast-follow][Folder picker] Use selectedItemValuePath from referenceParameters to get selection state